verifier/src/lib.rs

```diff
@@ -67,4 +69,66 @@ mod tests {
         let verifier = WormholeVerifier::new();
         verifier.verify(proof).unwrap();
     }
+
+    #[test]
+    fn cannot_verify_with_modified_exit_account() {
+        let prover = WormholeProver::new();
+        let inputs = CircuitInputs::default();
+        let mut proof = prover.commit(&inputs).unwrap().prove().unwrap();
+
+        println!("proof before: {:?}", proof.public_inputs);
+        let exit_account = ExitAccount::from_field_elements(&proof.public_inputs[16..20]);
+        println!("exit_account: {:?}", exit_account);
+        let modified_exit_account = ExitAccount::new([8u8; 32]);
+        proof.public_inputs[16..20].copy_from_slice(&modified_exit_account.to_field_elements());
+        println!("proof after: {:?}", proof.public_inputs);
+
+        // proof.public_inputs
+        let verifier = WormholeVerifier::new();
+        let result = verifier.verify(proof);
+        assert!(result.is_err(), "Expected proof to fail with modified exit_account");
+    }
+
+    #[test]
+    fn cannot_verify_with_any_public_input_modification() {
+        let prover = WormholeProver::new();
+        let inputs = CircuitInputs::default();
+        let mut proof = prover.commit(&inputs).unwrap().prove().unwrap();
+        let verifier = WormholeVerifier::new();
+
+        for ix in 0..proof.public_inputs.len() {
+            let mut p = proof.clone();
+            for jx in 0..8 {
+                p.public_inputs[ix].0 ^= 255 << 8 * jx;
+                let result = verifier.verify(p.clone());
+                assert!(result.is_err(), "Expected proof to fail with modified inputs");
+            }
+        }
+    }
+    
+    // #[test]
+    // fn cannot_verify_with_modified_proof() {
```

Comment:
We could just `#[ignore]` this, with the reason being that it takes too long, if we want to be able to run it but it not being run in CI. Similarly too the other fuzz test we have in `fuzz_tampered_proof`.